src/Relation/AbstractRelation.php

```diff
@@ -109,11 +109,7 @@ public function extract($data)
      */
     protected function isNullable(): bool
     {
-        if (array_key_exists(Relation::NULLABLE, $this->schema)) {
-            return $this->schema[Relation::NULLABLE];
-        }
-
-        return false;
+        return !empty($this->schema[Relation::NULLABLE]);
```

Comment:
Relation::NULLABLE must be configured with a bool value so again !(empty) will return `true` if set to true, false if set to false or not set.

There is also anothe rmatter about this: what if I set $this->schema[Relation::NULLABLE] to string "TRUE", "FALSE".
with !empty() it will return true (always a bool for empty or !empty), while previous code will return a string, forbidden by the type-hinted return value. BTW, if you'd want to allow "true", "false", "On", "Off" you could parse the value with filter_var FILTER_VALIDATE_BOOLEAN (like it's done in the php-ini settings)

Reply:
Perfect, thank you!